src/java/ognl/security/MethodBodyExecutionSandbox.java

```diff
@@ -0,0 +1,158 @@
+package ognl.security;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.Permissions;
+import java.security.Policy;
+import java.security.SecurityPermission;
+
+/**
+ * Guarantee a singleton shared thread-safe security manager and sandbox for user's methods body execution
+ * @author Yasser Zamani
+ * @since 3.1.23
+ */
+public class MethodBodyExecutionSandbox {
+    private static Permissions userDemandPermissions;
+    private static Policy userDemandPolicy;
+    private static SecurityManager userDemandSecurityManager;
+    private static boolean enabled;
+    private static boolean disabled;
+    private static Integer residentsCount = 0;
```

Comment:
It looks like no code needs residentsCount to be an "Integer" (object), might be worth making it a primitive "int" and avoid autoboxing overhead.

Reply:
:+1: Firstly I used it for sync but next I used .class but forgot to revert this. thanks! :)